Add build support for melodic

AdrianZw opened the pull request:
Resolves several issues on melodic, e.g. the class MoveGroup is removed in Melodic. Besides, i[ssue  #41 ](https://github.com/ros-industrial-consortium/reuleaux/issues/41) was resolved for me, since FindHDF5.cmake is used.
The branch needsto be tested on Kinetic and Noetic though.

AdrianZw committed:
Add build support for melodic

p0rys commented:
First, thank you for porting this package to melodic! 
I tried to build it with your commit and get the following error:

/catkin_ws/src/reuleaux/map_creator/include/map_creator/kinematics.h:9:10: fatal error: evoid_manipulator_ikfast_solver.cpp: Datei oder Verzeichnis nicht gefunden (File or Directory not found)
 #include "evoid_manipulator_ikfast_solver.cpp"

Is it possible that the commit is missing a new added file for the changes in Line 6 & 9 in _map_creator/include/map_creator/kinematics.h_ ?

AdrianZw commented:
Oh sorry, I forgot to remove this file from the CMakeLists.txt. The cpp is not needed. I was experimenting a bit and added the cpp file to the library. If you just remove it and you installed all the dependencies, it should compile. I will commit the changes as soon as possible. Thanks for pointing this out!Am 21.07.2020 12:19 schrieb p0rys <notifications@github.com>:
First, thank you for porting this package to melodic!
I tried to build it with your commit and get the following error:
/catkin_ws/src/reuleaux/map_creator/include/map_creator/kinematics.h:9:10: fatal error: evoid_manipulator_ikfast_solver.cpp: Datei oder Verzeichnis nicht gefunden (File or Directory not found)
#include "evoid_manipulator_ikfast_solver.cpp"
Is it possible that the commit is missing a new added file for the changes in Line 6 & 9 in map_creator/include/map_creator/kinematics.h ?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

AdrianZw committed:
Remove non default ikfast solver and add build fixes

sandakalum commented:
Thanks for adding build support for melodic, I was able to build on melodic. But when trying to add the Base Placement Plug-in panel in Rviz, it dies with 'Segmentation fault (core dumped)'. Any ideas?

marktkram commented:
Not sure if this is still active but your patch helped me a lot to get it running on melodic and should be merged :)
if you add qt5 support to the base_placment_plugin/CMakeLists.txt the plugin doesn't crash on startup.

`# Qt Stuff
if(rviz_QT_VERSION VERSION_LESS "5")
  find_package(Qt4 ${rviz_QT_VERSION} REQUIRED QtCore QtGui)
  include(${QT_USE_FILE})
  set(QT_LIBRARIES )
else()
  find_package(Qt5 ${rviz_QT_VERSION} REQUIRED Core Widgets Concurrent)
  set(QT_LIBRARIES Qt5::Core Qt5::Widgets Qt5::Concurrent)
endif()
# Instruct CMake to run moc automatically when needed.
set(CMAKE_AUTOMOC ON)`


`# Qt Stuff
if(rviz_QT_VERSION VERSION_LESS "5")
  qt4_wrap_cpp(base_placement_plugin_MOCS ${base_placement_plugin_HDRS} )
  qt4_wrap_ui(base_placement_plugin_UIS_H ${base_placement_plugin_UIS} )
else()
  qt5_wrap_cpp(base_placement_plugin_MOCS ${base_placement_plugin_HDRS} )
  qt5_wrap_ui(base_placement_plugin_UIS_H ${base_placement_plugin_UIS} )
endif()


include_directories(${base_placement_plugin_INCLUDE_DIRECTORIES} ${catkin_INCLUDE_DIRS} ${HDF5_INCLUDE_DIRS})
add_library(${PROJECT_NAME} ${base_placement_plugin_SRCS} ${base_placement_plugin_MOCS} ${base_placement_plugin_UIS_H}) # ${base_placement_plugin_UIS_H}
target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_LIBRARIES} yaml-cpp ${HDF5_LIBRARIES} create_marker)
## target_link_libraries(${PROJECT_NAME} yaml-cpp)`


Unfortunately i can't test this on kinetic

liyuping0906 commented:
@AdrianZw Hello, is there a noetic version of Reuleaux? I've tried compiling reuleaux on noetic but it hasn't worked. Thank you so much

AdrianZw commented:
Now it does, see https://github.com/AdrianZw/reuleaux/tree/noetic-devel/map_creator. At least it compiled and the map executor run on Ubuntu 22.04 with Noetic

I also added the fixes suggested by @marktkram 

I fixed a failing unit test, but I am not really sure if everything still works. Haven't used that tool in a longer time.

Is there a maintainer for this repo?

Not sure how to proceed with this...
